src/main/java/com/activestate/cloudfoundryjenkins/CloudFoundryPushPublisher.java

```diff
@@ -196,156 +217,172 @@ public boolean perform(AbstractBuild build, Launcher launcher, BuildListener lis
-                        if (instance.getState() == InstanceState.RUNNING) {
-                            running++;
-                        }
-                    }
-                    if (running == totalInstances && totalInstances > 0) {
-                        break;
-                    }
-                }
-                Thread.sleep(1000);
-            }
-
-            String instanceGrammar = "instances";
-            if (running == 1)
-                instanceGrammar = "instance";
-            listener.getLogger().println(running + " " + instanceGrammar + " running out of " + totalInstances);
-
-            if (running > 0) {
-                if (running != totalInstances) {
-                    listener.getLogger().println("WARNING: Some instances of the application are not running.");
-                }
-                if (deploymentInfo.isNoRoute()) {
-                    listener.getLogger().println("Application is now running. (No route)");
-                } else {
-                    listener.getLogger().println("Application is now running at " + appURI);
-                }
-                listener.getLogger().println("Cloud Foundry push successful.");
-                return true;
-            } else {
-                listener.getLogger().println("ERROR: The application failed to start after " + TIMEOUT + " seconds.");
-                listener.getLogger().println("Cloud Foundry push failed.");
-                return false;
-            }
-        } catch (CloudFoundryException e) {
-            listener.getLogger().println("ERROR: Unknown CloudFoundryException: " + e.getMessage());
-            listener.getLogger().println("ERROR: Cloud Foundry error code: " + e.getCloudFoundryErrorCode());
-            if (e.getDescription() != null) {
-                listener.getLogger().println("ERROR: " + e.getDescription());
-            }
-            e.printStackTrace(listener.getLogger());
-            return false;
-        } catch (FileNotFoundException e) {
-            listener.getLogger().println("ERROR: Could not find file: " + e.getMessage());
-            return false;
-        } catch (ZipException e) {
-            listener.getLogger().println("ERROR: ZipException: " + e.getMessage());
-            return false;
-        }
-    }
+ BuildListener listener, URL targetUrl) throws IOException,
+			InterruptedException {
+		try {
+			String OrigAppURI = "https://" + deploymentInfo.getHostName() + "." + deploymentInfo.getDomain();
+			System.out.println("The Orig Route  "+OrigAppURI);
+			listener.getLogger().println("Pushing " + deploymentInfo.getAppName() + " app to " + target);
+
+			// This is where we would create services, if we decide to add that feature.
+			// List<CloudService> cloudServices = deploymentInfo.getServices();
+			// client.createService();
+			String greenDeployAppName = handleExistingAppAndUpdateDeploymentInfo(client, listener, deploymentInfo);
```

Comment:
The method handleExistingAppAndUpdateDeploymentInfo should not return anything. Its already doing two things - why the third one?
The app name can be obtained from deploymentInfo